core/src/test/java/com/linecorp/armeria/client/limit/ConcurrencyLimitingClientTest.java

```diff
@@ -216,28 +243,84 @@ void testUnlimitedRequestWithFaultyDelegate() throws Exception {
         await().untilAsserted(() -> assertThat(client.numActiveRequests()).isZero());
     }
 
-    private static ClientRequestContext newContext() {
-        return ClientRequestContext.builder(HttpRequest.of(HttpMethod.GET, "/"))
-                                   .eventLoop(eventLoop.get())
-                                   .build();
+    @Test
+    void configuresClientWithConcurrencyLimit() {
+        final ConcurrencyLimit concurrencyLimit = ConcurrencyLimit.builder()
+                                                                  .maxConcurrency(1)
+                                                                  .timeout(Duration.of(10, SECONDS))
+                                                                  .policy(requestContext -> true)
+                                                                  .build();
+        final ConcurrencyLimitingClient client = newDecorator(concurrencyLimit).apply(delegate);
+        assertThat(client.concurrencyLimit()).isEqualTo(concurrencyLimit);
     }
 
-    /**
-     * Closes the response returned by the delegate and consumes everything from it, so that its close future
-     * is completed.
-     */
-    private static void closeAndDrain(HttpResponseWriter actualRes, HttpResponse deferredRes) {
-        actualRes.close();
-        deferredRes.subscribe(NoopSubscriber.get());
-        deferredRes.whenComplete().join();
-        waitForEventLoop();
-    }
+    @Test
+    void skipsConcurrencyBasedOnRequestContext() throws Exception {
+        final ClientRequestContext ctx = newContext();
+        final HttpRequest req = newReq();
+        final HttpResponseWriter actualRes = HttpResponse.streaming();
 
-    private static void waitForEventLoop() {
-        eventLoop.get().submit(() -> { /* no-op */ }).syncUninterruptibly();
+        when(delegate.execute(ctx, req)).thenReturn(actualRes);
+        final ConcurrencyLimit concurrencyLimit = ConcurrencyLimit.builder()
+                                                                  .maxConcurrency(1)
```

Comment:
If this is for testing the policy which always returns `false`, shouldn't we try sending the requests more than 1?